cmd/ndm_daemonset/probe/addhandler.go

```diff
@@ -54,6 +63,36 @@ func (pe *ProbeEvent) addBlockDevice(bd blockdevice.BlockDevice, bdAPIList *apis
 
 	pe.addBlockDeviceToHierarchyCache(bd)
 
+	// handle devices that are not managed by NDM
+	// eg:devices in use by mayastor, zfs PV and jiva
+	// TODO jiva handling is still to be added.
+	if ok, err := pe.handleUnmanagedDevices(bd, bdAPIList); err != nil {
+		klog.Errorf("error handling unmanaged device %s. error: %v", bd.DevPath, err)
+		return err
+	} else if !ok {
+		klog.V(4).Infof("processed device: %s being used by mayastor/zfs-localPV", bd.DevPath)
+		return nil
+	}
+
+	// if parent device in use, no need to process further
+	if ok, err := pe.isParentDeviceInUse(bd); err != nil {
+		klog.Error(err)
+		return err
+	} else if ok {
+		klog.Infof("parent device of device: %s in use", bd.DevPath)
+		return nil
+	}
+
+	// upgrades the devices that are in use and used the legacy method
+	// for uuid generation.
+	if ok, err := pe.upgradeBD(bd, bdAPIList); err != nil {
+		klog.Errorf("upgrade of device: %s failed. Error: %v", bd.DevPath, err)
+		return err
+	} else if !ok {
+		klog.V(4).Infof("device: %s upgraded", bd.DevPath)
+		return nil
+	}
+
```

Comment:
see if we can break addBlockDevice into smaller part?

Reply:
will take it up in a later PR @pawanpraka1 .